packages/pyright-internal/src/analyzer/typeEvaluatorWithTracker.ts

```diff
@@ -25,141 +25,46 @@ export function createTypeEvaluatorWithTracker(
     logger: LogTracker,
     printer?: TracePrinter
 ) {
-    if (!evaluatorOptions.logCalls && isDebugMode()) {
-        return createTypeEvaluator(importLookup, evaluatorOptions);
+    function wrapWithLogger<T extends (...args: any[]) => any>(func: T): (...args: Parameters<T>) => ReturnType<T> {
+        // Only wrap the function if told to do so and the log level is high enough for it
+        // to actually log something
+        if (evaluatorOptions.logCalls && logger.logLevel === LogLevel.Log) {
+            return (...args: Parameters<T>): ReturnType<T> => {
+                return logger.log(
+                    func.name,
+                    (s) => {
+                        if (func.name === 'importLookup' && args.length > 0) {

```

Comment:
The alternative is to have a different function factory for just importLookup. Or maybe something else I'm not seeing. 

Ideally the tracePrinter would be able to distinguish between a path and a string, but figured if that became a problem, it would only be a part of logging and not something that would affect most users.

Reply:
I believe the worst case scenario here is that the path isn't printed with the full path and just ends up printing the subpath.

See the code here:
https://github.com/rchiodo/pyright/blob/75878a84b8be940f17afeb00e16cf4213921e45a/packages/pyright-internal/src/analyzer/tracePrinter.ts#L42

Which instead of doing that, it would run the plain print code.

Reply:
What if someone happens to change the signature of `importLookup`? I guess in that unlikely case, it will likely cause an immediately-noticeable crash.

Reply:
No I think it would just call the printFileOrModuleName with the first arg. I'll add some protection there to make sure it doesn't crash.